addons/google_gmail/models/ir_mail_server.py

```diff
@@ -12,6 +13,17 @@ class IrMailServer(models.Model):
     _name = 'ir.mail_server'
     _inherit = ['ir.mail_server', 'google.gmail.mixin']
 
+    @api.constrains('use_google_gmail_service')
+    def _check_use_google_gmail_service(self):
+        for server in self:
+            if not server.use_google_gmail_service:
+                continue
+
+            if not self.smtp_user:
```

Comment:
```suggestion
            if not server.smtp_user:
```
Good catch, classic copy pasta blunder. I need more coffee ;)